crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -1080,7 +1118,16 @@ impl<InstalledPackages: InstalledPackagesProvider> ResolverState<InstalledPackag
         };
 
         // Check whether the version is incompatible due to its Python requirement.
-        if let Some(incompatibility) = Self::check_requires_python(dist, python_requirement) {
+        if let Some((requires_python, incompatibility)) =
+            Self::check_requires_python(dist, python_requirement)
+        {
+            if env.marker_environment().is_none() {
+                let forks = VersionForker::fork(requires_python, python_requirement, env);
+                if !forks.is_empty() {
```

Comment:
maybe something that we handle implicitly, but wouldn't we always return at least one fork (because union(left, right) is the entire requires-python of the fork we're in, so partitioning it at least one partition must overlap), so when we get one fork we should also continue implicitly?

Reply:
I had a similar thought here. But I find it somewhat difficult to reason about invariants that are always true. For example, if the fork has a marker that always evaluates to `false`, then `VersionForker::fork` could return an empty set of forks. But... I think it's impossible at this point to have such a marker?

Other than that, I think I agree with you. We could add an `assert!` here.